modules/Stories/Stories.tsx

```diff
@@ -1,27 +1,46 @@
+import type { Category } from '@prezly/sdk';
 import type { Locale } from '@prezly/theme-kit-nextjs';
 
 import { app } from '@/adapters/server';
 
 import { InfiniteStories } from '../InfiniteStories';
 
 interface Props {
+    categoryId: Category['id'] | undefined;
     localeCode: Locale.Code;
     pageSize: number;
 }
 
-export async function Stories({ localeCode, pageSize }: Props) {
+export async function Stories({ categoryId, localeCode, pageSize }: Props) {
     const newsroom = await app().newsroom();
     const languageSettings = await app().languageOrDefault(localeCode);
+    const categories = await app().categories();
+    const featuredCategories = categories.filter(
+        ({ is_featured, i18n }) => is_featured && i18n[localeCode]?.public_stories_number > 0,
+    );
+    const hasFeaturedCategories = featuredCategories.length > 0;
+
+    const { stories: pinnedOrMostRecentStories } = await app().stories({
+        limit: 1,
+        locale: { code: localeCode },
+    });
+
     const { stories, pagination } = await app().stories({
+        category: categoryId ? { id: categoryId } : undefined,
         limit: pageSize,
         locale: { code: localeCode },
     });
 
     return (
         <InfiniteStories
+            key={categoryId}
+            category={categoryId ? { id: categoryId } : undefined}
+            categories={featuredCategories}
             newsroomName={languageSettings.company_information.name || newsroom.name}
             pageSize={pageSize}
-            initialStories={stories}
+            initialStories={
+                hasFeaturedCategories ? [...pinnedOrMostRecentStories, ...stories] : stories
```

Comment:
This change is required to keep the pinned/most recent story static when you start filtering, so only the 2nd story and onwards changes.

The downside is that the pinned/most recent story can be potentially duplicated in the filtered stories below, but we couldn't think of a simple way to solve this that would not incorporate fiddling with the pagination.